packages/braket/quri_parts/braket/backend/transpiler.py

```diff
@@ -26,10 +26,22 @@ class AwsDeviceTranspiler(SequentialTranspiler):
     """
 
     def __init__(self, device: AwsDevice):
+        name_action_type = {
+            "LocalSimulator": "braket.ir.jaqcd.program",
+            "SV1": "braket.ir.jaqcd.program",
+            "TN1": "braket.ir.jaqcd.program",
+            "dm1": "braket.ir.jaqcd.program",
+            "Harmony": "braket.ir.jaqcd.program",
+            "Aria 1": "braket.ir.openqasm.program",
+            "Aria 2": "braket.ir.openqasm.program",
+            "Garnet": "braket.ir.openqasm.program",
+        }
```

Comment:
I think having this kind of mapping is fragile to updates on their platform (e.g. addition of new devices).
Another more robust approach may be to check `braket.ir.openqasm.program` first and then `braket.ir.jaqcd.program`.
(Looking at [the current definition of DeviceActionType](https://amazon-braket-schemas-python.readthedocs.io/en/latest/_apidoc/braket.device_schema.device_action_properties.html#braket.device_schema.device_action_properties.DeviceActionType), only those two seem to be compatible with the gate model)

Reply:
Right.. I believe this fixes it!